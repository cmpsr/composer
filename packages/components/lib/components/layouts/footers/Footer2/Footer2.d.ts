import { ReactElement } from "react";
import { FooterStyle } from "types";
export declare const FOOTER2_DEFAULT_TEST_ID = "footer2";
declare type Props = {
    copy?: string;
    className?: string;
    children?: {
        logo?: ReactElement;
        secondaryContent?: ReactElement | ReactElement[];
    };
    style: FooterStyle;
    testId?: string;
};
export declare const Footer2: ({ className, copy, children, style, testId, }: Props) => JSX.Element;
export {};
//# sourceMappingURL=Footer2.d.ts.map