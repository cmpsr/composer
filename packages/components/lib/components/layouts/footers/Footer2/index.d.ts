export * from './Footer2';
//# sourceMappingURL=index.d.ts.map