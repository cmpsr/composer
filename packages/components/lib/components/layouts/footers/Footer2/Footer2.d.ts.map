{"version":3,"file":"Footer2.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/footers/Footer2/Footer2.tsx"],"names":[],"mappings":"AAAA,OAAc,EAAE,YAAY,EAAE,MAAM,OAAO,CAAC;AAM5C,OAAO,EAAE,WAAW,EAAE,MAAM,OAAO,CAAC;AAEpC,eAAO,MAAM,uBAAuB,YAAY,CAAC;AAEjD,aAAK,KAAK,GAAG;IACX,IAAI,CAAC,EAAE,MAAM,CAAC;IACd,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,QAAQ,CAAC,EAAE;QACT,IAAI,CAAC,EAAE,YAAY,CAAC;QACpB,gBAAgB,CAAC,EAAE,YAAY,GAAG,YAAY,EAAE,CAAC;KAClD,CAAC;IACF,KAAK,EAAE,WAAW,CAAC;IACnB,MAAM,CAAC,EAAE,MAAM,CAAC;CACjB,CAAC;AAEF,eAAO,MAAM,OAAO,kDAMjB,KAAK,gBA4BP,CAAC"}