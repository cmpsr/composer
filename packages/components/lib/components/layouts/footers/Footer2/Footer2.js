"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Footer2 = exports.FOOTER2_DEFAULT_TEST_ID = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _containers = require("../../../containers");

var _utils = require("../../../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FOOTER2_DEFAULT_TEST_ID = "footer2";
exports.FOOTER2_DEFAULT_TEST_ID = FOOTER2_DEFAULT_TEST_ID;

const Footer2 = ({
  className,
  copy,
  children,
  style,
  testId = FOOTER2_DEFAULT_TEST_ID
}) => {
  const colorClasses = (0, _utils.getClassesFromStyle)(style.background);
  return /*#__PURE__*/_react.default.createElement(_containers.Box, {
    size: "full",
    className: (0, _classnames.default)("py-8", className, colorClasses),
    testId: testId
  }, /*#__PURE__*/_react.default.createElement(_containers.Box, {
    size: "content",
    className: "items-center flex flex-col md:flex-row"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1 flex justify-start"
  }, children && children.logo), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1 flex w-full text-center my-8 md:my-0 justify-center"
  }, children && children.secondaryContent && /*#__PURE__*/_react.default.createElement(_containers.Grid, {
    className: (0, _classnames.default)("w-full md:w-auto gap-8 md:gap-24 mx-4 md:mx-0", `grid-cols-${_react.default.Children.count(children.secondaryContent)}`)
  }, children.secondaryContent)), /*#__PURE__*/_react.default.createElement("p", {
    className: "flex-1 text-center md:text-right my-4 md:my-0"
  }, copy)));
};

exports.Footer2 = Footer2;