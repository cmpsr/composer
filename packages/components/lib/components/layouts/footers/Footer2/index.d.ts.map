{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/footers/Footer2/index.ts"],"names":[],"mappings":"AAAA,cAAc,WAAW,CAAC"}