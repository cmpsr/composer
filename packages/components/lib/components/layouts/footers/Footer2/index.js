"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Footer = require("./Footer2");

Object.keys(_Footer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Footer[key];
    }
  });
});