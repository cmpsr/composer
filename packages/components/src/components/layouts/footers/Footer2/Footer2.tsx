import React, { ReactElement } from "react";
import cn from "classnames";

import { Box, Grid } from "components/containers";
import { getClassesFromStyle } from "utils";

import { FooterStyle } from "types";

export const FOOTER2_DEFAULT_TEST_ID = "footer2";

type Props = {
  copy?: string;
  className?: string;
  children?: {
    logo?: ReactElement;
    secondaryContent?: ReactElement | ReactElement[];
  };
  style: FooterStyle;
  testId?: string;
};

export const Footer2 = ({
  className,
  copy,
  children,
  style,
  testId = FOOTER2_DEFAULT_TEST_ID,
}: Props) => {
  const colorClasses = getClassesFromStyle(style.background);
  return (
    <Box
      size="full"
      className={cn("py-8", className, colorClasses)}
      testId={testId}
    >
      <Box size="content" className="items-center flex flex-col md:flex-row">
        <div className="flex-1 flex justify-start">
          {children && children.logo}
        </div>
        <div className="flex-1 flex w-full text-center my-8 md:my-0 justify-center">
          {children && children.secondaryContent && (
            <Grid
              className={cn(
                "w-full md:w-auto gap-8 md:gap-24 mx-4 md:mx-0",
                `grid-cols-${React.Children.count(children.secondaryContent)}`
              )}
            >
              {children.secondaryContent}
            </Grid>
          )}
        </div>
        <p className="flex-1 text-center md:text-right my-4 md:my-0">{copy}</p>
      </Box>
    </Box>
  );
};
