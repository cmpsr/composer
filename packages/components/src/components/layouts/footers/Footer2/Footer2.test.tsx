import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Footer2, FOOTER2_DEFAULT_TEST_ID } from './Footer2';

describe('Footer2', () => {
  const testStyle = { background: { default: { bg: 'white' } } };
  it('should match snapshot', () => {
    const footer = renderer.create(<Footer2 style={testStyle} />);
    const tree = footer.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(<Footer2 style={testStyle} />);
    const footer = getByTestId(FOOTER2_DEFAULT_TEST_ID);
    expect(footer).toBeInTheDocument();
  });
  it('should render style', () => {
    const { getByTestId } = render(<Footer2 style={testStyle} />);
    const footer = getByTestId(FOOTER2_DEFAULT_TEST_ID);
    expect(footer.classList).toContain('bg-white');
  });
  it('should render copy', () => {
    const testCopy = 'Test copy';
    const { getByText } = render(<Footer2 style={testStyle} copy={testCopy} />);
    const copy = getByText(testCopy);
    expect(copy).toBeInTheDocument();
  });
  it('should render children logo', () => {
    const childrenLogoTestId = 'Children-Logo-TestId';
    const renderedChildren = {
      logo: <p data-testid={childrenLogoTestId}>Logo</p>,
      content: <div>Children</div>,
    };
    const { getByTestId } = render(
      <Footer2 style={testStyle}>{renderedChildren}</Footer2>,
    );
    const childrenLogo = getByTestId(childrenLogoTestId);
    expect(childrenLogo).toBeInTheDocument();
    expect(childrenLogo.parentElement.classList).toContain('flex-1');
    expect(childrenLogo.parentElement.classList).toContain('flex');
    expect(childrenLogo.parentElement.classList).toContain('justify-start');
  });
  it('should render children with appropriate style', () => {
    const childrenTestId = 'Children-TestId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      secondaryContent: <div data-testid={childrenTestId}>Children</div>,
    };
    const { baseElement, getByTestId } = render(
      <Footer2 style={testStyle}>{renderedChildren}</Footer2>,
    );
    const children = getByTestId(childrenTestId);
    expect(children).toBeInTheDocument();
    expect(baseElement.innerHTML).toContain('grid-cols-1');
  });
  it('should render array of children with appropriate style', () => {
    const childrenTestId = 'Children-TestId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      secondaryContent: [
        <div data-testid={childrenTestId + '1'} key="1">
          Children1
        </div>,
        <div data-testid={childrenTestId + '2'} key="2">
          Children2
        </div>,
      ],
    };
    const { baseElement, getAllByTestId } = render(
      <Footer2 style={testStyle}>{renderedChildren}</Footer2>,
    );
    const children = getAllByTestId(childrenTestId, { exact: false });
    expect(children.length).toBe(2);
    expect(baseElement.innerHTML).toContain('grid-cols-2');
  });
});
