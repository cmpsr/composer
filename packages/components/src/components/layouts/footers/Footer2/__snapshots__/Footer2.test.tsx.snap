// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer2 should match snapshot 1`] = `
<div
  className="w-full h-full py-8 bg-white"
  data-testid="footer2"
>
  <div
    className="w-full h-full max-w-content mx-auto items-center flex flex-col md:flex-row"
    data-testid="box"
  >
    <div
      className="flex-1 flex justify-start"
    />
    <div
      className="flex-1 flex w-full text-center my-8 md:my-0 justify-center"
    />
    <p
      className="flex-1 text-center md:text-right my-4 md:my-0"
    />
  </div>
</div>
`;
